api/src/components/reference/__tests__/updateReference.test.ts

```diff
@@ -0,0 +1,83 @@
+import * as testUtils from 'lib/testUtils';
+
+describe('update a reference', () => {
+    beforeEach(async () => {
+        await testUtils.clearDB();
+        await testUtils.testSeed();
+    });
+
+    test('An authenticated author can update a reference', async () => {
+        const reference = await testUtils.agent
+            .patch('/publications/publication-interpretation-draft/reference/01')
+            .query({ apiKey: '123456789' })
+            .send({
+                id: '01',
+                location: 'http://octopus.ac',
+                publicationId: 'publication-problem-draft',
+                text: '<p>Reference 1</p>',
+                type: 'TEXT'
+            });
+
+        expect(reference.status).toEqual(200);
+    });
+
+    test('The publication must be in draft state', async () => {
+        const reference = await testUtils.agent
+            .patch('/publications/publication-real-world-application-live/reference/03')
+            .query({ apiKey: '123456789' })
+            .send({
+                id: '03',
+                location: 'http://octopus.ac',
+                publicationId: 'publication-real-world-application-live',
+                text: '<p>Reference 1</p>',
+                type: 'TEXT'
+            });
+
+        expect(reference.status).toEqual(403);
+    });
+
+    test('The user must be the author of the publication', async () => {
+        const reference = await testUtils.agent
+            .patch('/publications/publication-interpretation-draft/reference/01')
+            .query({ apiKey: '000000005' })
+            .send({
+                id: '01',
+                location: 'http://octopus.ac',
+                publicationId: 'publication-problem-draft',
+                text: '<p>Reference 1</p>',
+                type: 'TEXT'
+            });
+
+        expect(reference.status).toEqual(403);
+    });
+
+    test('The id cannot be amended', async () => {
+        const reference = await testUtils.agent
+            .patch('/publications/publication-interpretation-draft/reference/01')
+            .query({ apiKey: '000000005' })
+            .send({
+                id: '0000004',
+                location: 'http://octopus.ac',
+                publicationId: 'publication-problem-draft',
+                text: '<p>Reference 1</p>',
+                type: 'TEXT'
+            });
+
+        expect(reference.status).toEqual(403);
+    });
+
+    test('The publication id cannot be amended', async () => {
+        const reference = await testUtils.agent
+            .patch('/publications/publication-interpretation-draft/reference/01')
+            .query({ apiKey: '000000005' })
+            .send({
+                id: '01',
+                location: 'http://octopus.ac',
+                publicationId: 'made-up-publication-id',
```

Comment:
I think this only works because 'made-up-publication-id' does not exist (so prisma/db fails), if you used  'publication-problem-draft' it would update it,  incorrectly

Reply:
actually it's the ownership code that is being hit, if you log the reference returned you see 

text: '{"message":"User is not the author of this publication."}',

Reply:
This is returning 403 because of the API key(this user is not the author). Not really testing if the publicationId can be changed or not.

To be honest, I don't think we should bother testing all these endpoints because we only need GET ALL and UPDATE ALL for references. And that's all we should test. We can safely remove the rest of the routes and handlers for references. If we need anything else in the future, we can simply implement it.